pkgs/applications/office/jabref/default.nix

```diff
@@ -1,73 +1,144 @@
-        --set-rpath "${ lib.makeLibraryPath [ zlib ]}:$out/lib/runtime/lib:$out/lib/runtime/lib/server" $f
-    done
+    # script to support browser extensions
+    install -Dm755 buildres/linux/jabrefHost.py $out/lib/jabrefHost.py
+    # This can be removed in the next version
+    sed -i -e "/importBibtex/s/{}/'{}'/" $out/lib/jabrefHost.py
+    install -Dm644 buildres/linux/native-messaging-host/firefox/org.jabref.jabref.json $out/lib/mozilla/native-messaging-hosts/org.jabref.jabref.json
+    sed -i -e "s|/opt/jabref|$out|" $out/lib/mozilla/native-messaging-hosts/org.jabref.jabref.json
 
-    for f in $out/lib/runtime/lib/*.so; do
-      patchelf \
-        --set-rpath "${systemLibPaths}:$out/lib/runtime/lib:$out/lib/runtime/lib/server" $f
-    done
+    # Resources in the jar can't be found, workaround copied from AUR
+    cp -r build/resources $out/share/java/jabref
 
-    # patching the libs in the JImage runtime image is quite impossible as there is no documented way
-    # of rebuilding the image after it has been extracted
-    # the image format itself is "intendedly not documented" - maybe one of the reasons the
-    # devolpers constantly broke "jimage recreate" and dropped it in OpenJDK 9 Build 116 Early Access
-    # so, for now just copy the image and provide our lib paths through the wrapper
+    # workaround for https://github.com/NixOS/nixpkgs/issues/162064
+    tar xf build/distributions/JabRef-${version}.tar -C $out --strip-components=1
+    unzip $out/lib/javafx-web-18-linux${lib.optionalString stdenv.isAarch64 "-aarch64"}.jar libjfxwebkit.so -d $out/lib/
 
-    makeWrapper $out/lib/runtime/bin/java $out/bin/jabref \
-      --add-flags '-Djava.library.path=${systemLibPaths}' --add-flags "-p $out/lib/app -m org.jabref/org.jabref.JabRefLauncher" \
-      --prefix LD_LIBRARY_PATH : '${systemLibPaths}'
+    wrapProgram $out/bin/JabRef \
+      --prefix PATH : ${lib.makeBinPath [ xdg-utils ]} \
+      --set JAVA_HOME "${jdk}" \
+      --set JAVA_OPTS "-Djava.library.path=$out/lib/ --patch-module org.jabref=$out/share/java/jabref/resources/main"
 
-    cp -r ${desktopItem}/share/applications $out/share/
+    # lowercase alias (for convenience and required for browser extensions)
+    ln -sf $out/bin/JabRef $out/bin/jabref
 
-    # we still need to unpack the runtime image to get the icon
-    mkdir unpacked
-    $out/lib/runtime/bin/jimage extract --dir=./unpacked lib/runtime/lib/modules
-    cp unpacked/org.jabref/icons/jabref.svg $out/share/icons/jabref.svg
+    runHook postInstall
   '';
 
   meta = with lib; {
-    broken = (stdenv.isLinux && stdenv.isAarch64);
     description = "Open source bibliography reference manager";
     homepage = "https://www.jabref.org";
+    sourceProvenance = with sourceTypes; [
+      fromSource
+      binaryBytecode  # source bundles dependencies as jars
+      binaryNativeCode  # source bundles dependencies as jars
+    ];
```

Comment:
You can try remove these lines, we don't have https://github.com/NixOS/nixpkgs/pull/161098 in 22.05

Reply:
Thanks for mentioning, with this all checks are green.